cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/loadbalancer/GcpForwardingRuleResourceBuilder.java

```diff
@@ -0,0 +1,131 @@
+import com.sequenceiq.cloudbreak.cloud.gcp.network.GcpSubnetResourceBuilder;
+import com.sequenceiq.cloudbreak.cloud.model.CloudLoadBalancer;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.CloudStack;
+import com.sequenceiq.cloudbreak.cloud.model.TargetGroupPortPair;
+import com.sequenceiq.common.api.type.LoadBalancerType;
+import com.sequenceiq.common.api.type.ResourceType;
+
+public class GcpForwardingRuleResourceBuilder extends AbstractGcpLoadBalancerBuilder {
+
+    public static final String GCP_BACKEND_SERVICE_REF_FORMAT = "https://www.googleapis.com/compute/v1/projects/%s/regions/%s/backendServices/%s";
+
+    public static final String GCP_IP_REF_FORMAT = "https://www.googleapis.com/compute/v1/projects/%s/regions/%s/addresses/%s";
+
+    private static final int ORDER = 4;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GcpForwardingRuleResourceBuilder.class);
+
+    @Override
+    public List<CloudResource> create(GcpContext context, AuthenticatedContext auth, CloudLoadBalancer loadBalancer) {
+        List<CloudResource> resources = new ArrayList<>();
+        for (TargetGroupPortPair targetGroupPortPair : loadBalancer.getPortToTargetGroupMapping().keySet()) {
+            Integer healthCheckPort = targetGroupPortPair.getHealthCheckPort();
+            String resourceName = getResourceNameService().resourceName(resourceType(), context.getName(),
+                    loadBalancer.getType(), targetGroupPortPair.getTrafficPort());
+            Map<String, Object> parameters = Map.of(TRAFFICPORT, targetGroupPortPair.getTrafficPort(), HCPORT, targetGroupPortPair.getHealthCheckPort());
+            resources.add(new CloudResource.Builder().type(resourceType())
+                    .name(resourceName)
+                    .params(parameters)
+                    .build());
+        }
+        return resources;
+    }
+
+    @Override
+    public List<CloudResource> build(GcpContext context, AuthenticatedContext auth, List<CloudResource> buildableResources,
+            CloudLoadBalancer loadBalancer, CloudStack cloudStack) throws Exception {
+        String projectId = context.getProjectId();
+        String regionName = context.getLocation().getRegion().getRegionName();
+        List<CloudResource> results = new ArrayList<>();
+
+
+        List<CloudResource> backendResources = filterResourcesByType(context.getLoadBalancerResources(loadBalancer.getType()), ResourceType.GCP_BACKEND_SERVICE);
+        List<CloudResource> ipResources = filterResourcesByType(context.getLoadBalancerResources(loadBalancer.getType()), ResourceType.GCP_RESERVED_IP);
+
+        for (CloudResource buildableResource : buildableResources) {
+            Integer trafficPort = buildableResource.getParameter(TRAFFICPORT, Integer.class);
+            ForwardingRule forwardingRule = new ForwardingRule().setIPProtocol("TCP")
```

Comment:
I generally think string literals should be constants instead of hardcoded wherever possible. That makes them easy to find in the scenario where they might have to be updated, and easier to share if we eventually find we're hardcoding the same values multiple places.